src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -340,8 +340,8 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
 
                 if (connectionStatus.Type == SubscriptionConnectionServerMessage.MessageType.Error)
                     message += $". Exception: {connectionStatus.Exception}";
-
-                throw new Exception(message);
+                
+                throw new SubscriptionMessageTypeException(message);
```

Comment:
need to handle it in `ShouldTryToReconnect`, no?

Reply:
Fixed